### PR TITLE
Refreshed existing quickstart CFN templates and add private endpoint and activation

### PR DESCRIPTION
**Pull request by SuperMohit**
*Description of changes:*
1. MongoDB has published the latest Atlas CFN resources in multiple AWS Regions. As a result, these quickstarts need to meet the schema constraints of the resources.
2. Activation resource is also added to allow for activating resources needed in the quickstart.
3. Private Endpoint Quickstart has also been added.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/mongodb-atlas.private-endpoint.template.yaml [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+      DatabaseUserRoleDatabaseName:
+        default: MongoDB Atlas Database User Role Database Name
+      ActivateMongoDBResources:
+        default: Activate MongoDB Atlas CloudFormation resources
+      VPCCIDR:
+        default: VPC CIDR.
+      PrivateSubnetCIDR:
+          default: Private subnet CIDR
+      AvailabilityZone:
+        deafault: AZ fo Subnet
```
Comment (vsnyc) [inline templates/mongodb-atlas.private-endpoint.template.yaml:60]: Typo and indentation issue.
```suggestion
       default: AZ for subnet
```

File: templates/mongodb-atlas.base-template.yaml [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+      ClusterInstanceSize:
+        default: MongoDB Atlas Instance Size
+      DatabaseUserRoleDatabaseName:
+        default: MongoDB Atlas Database User Role Database Name
+      VPCCIDR:
+        default: VPC CIDR.
+      PrivateSubnetCIDR:
+        default: Private subnet CIDR
+      AvailabilityZone:
+         deafault: AZ fo Subnet
```
Comment (vsnyc) [inline templates/mongodb-atlas.base-template.yaml:48]: Typo and indentation issue.
```suggestion
       default: AZ for subnet
```

File: templates/mongodb-atlas.private-endpoint.template.yaml [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+      ClusterInstanceSize:
+        default: MongoDB Atlas Instance Size
+      DatabaseUserRoleDatabaseName:
+        default: MongoDB Atlas Database User Role Database Name
+      ActivateMongoDBResources:
+        default: Activate MongoDB Atlas CloudFormation resources
+      VPCCIDR:
+        default: VPC CIDR.
+      PrivateSubnetCIDR:
+          default: Private subnet CIDR
```
Comment (vsnyc) [inline templates/mongodb-atlas.private-endpoint.template.yaml:58]: Fix indentation
```suggestion
        default: Private subnet CIDR
```

File: templates/mongodb-atlas.private-endpoint.template.yaml [hunk 4]
```diff
@@ -0,0 +1,267 @@
+---
+AWSTemplateFormatVersion: '2010-09-09'
+Description: MongoDB Atlas AWS CloudFormation Quickstart with VPC Peering. (qs-1rkorhefe)
```
Comment (vsnyc) [inline templates/mongodb-atlas.private-endpoint.template.yaml:3]: Can't reuse template IDs. Replace with `qs-1tq8f0hrt`
```suggestion
Description: MongoDB Atlas AWS CloudFormation Quickstart with VPC Peering. (qs-1tq8f0hrt)
```

File: templates/activate-mongodb-atlas-resources.yaml [hunk 5]
```diff
@@ -0,0 +1,155 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: MongoDB Resource activation. (qs-1rkorhec7)
```
Comment (vsnyc) [inline templates/activate-mongodb-atlas-resources.yaml:2]: Can't reuse template IDs. Replace with `qs-1tq8f0hon`.
```suggestion
Description: MongoDB Resource activation. (qs-1tq8f0hon)
```

File: templates/mongodb-atlas.base-template.yaml [hunk 6]
```diff
@@ -0,0 +1,329 @@
+---
+AWSTemplateFormatVersion: '2010-09-09'
+Description: MongoDB Atlas AWS CloudFormation Quickstart. (qs-1rkorhec7)
```
Comment (vsnyc) [inline templates/mongodb-atlas.base-template.yaml:3]: Can't reuse template IDs. Replace with: `qs-1tq8f0hrl`.
```suggestion
Description: MongoDB Atlas AWS Partner Solution. (qs-1tq8f0hrl))
```

File: templates/mongodb-atlas.base-template.yaml [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+      - "EU_NORTH_1"
+      - "EU_WEST_2"
+      - "EU_WEST_3"
+      - "EU_SOUTH_1"
+      - "ME_SOUTH_1"
+      - "AF_SOUTH_1"
+  ClusterMongoDBMajorVersion:
+    Description: The version of MongoDB
+    Type: String
+    Default: "4.4"
```
Comment (vsnyc) [inline templates/mongodb-atlas.base-template.yaml:131]: Can't we default to the latest version and removed old version that customers should no longer use from the `AllowedValues` list?

File: templates/mongodb-atlas-main.template.yaml [hunk 8]
```diff
@@ -184,12 +183,11 @@ Resources:
     Type: AWS::CloudFormation::Stack
     Properties:
       TemplateURL: !Sub
-        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/register-mongodb-atlas-resources.yaml'
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/activate-mongodb-atlas-resources.yaml'
```
Comment (vsnyc) [inline templates/mongodb-atlas-main.template.yaml:186]: We should rename the resource from `RegisterAtlasResources` to `ActivateAtlasResources` for consistency. The condition should be renamed as well.

**Review by vsnyc — CHANGES_REQUESTED** [review]
Added review comments inline. In addition, do the following changes:

1. Rename `activate-mongodb-atlas-resources.yaml` to `activate-mongodb-atlas-resources.template.yaml`.
2. Rename `mongodb-atlas.base-template.yaml` to `mongodb-atlas.base.template.yaml`.
3. Delete `register-mongodb-atlas-resources.yaml` - the file is no longer used.
4. Can we add a test for the private endpoint configuration?
5. Also, could you test the code with taskcat and attach a zip of the `taskcat_outputs` demonstrating a successful test run?

After that the submodule dependencies need to be updated, I can fix that.

File: templates/mongodb-atlas.private-endpoint.template.yaml [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+      - "eu-central-1"
+      - "eu-west-1"
+      - "eu-north-1"
+      - "eu-west-1"
+      - "eu-west-2"
+      - "eu-west-3"
+      - "eu-south-1"
+      - "me-south-1"
+      - "af-south-1"
+  VPCCIDR:
```
[the remark this answers is not in the document]
Comment (SuperMohit) [reply templates/mongodb-atlas.private-endpoint.template.yaml:180]: Yes, for creating VPC

**Comment by vsnyc** [comment]
@SuperMohit - I am not seeing your changes for the resolved issues? Have you pushed them yet?

**Comment by SuperMohit** [comment]
@vsnyc I have resolved most of the comments. I need to update taskcat.

**Comment by SuperMohit** [comment]
https://quickstart-pi.s3.amazonaws.com/quickstart-temp/tCaT-mdb-atlas-pvt-endpoint-658e92-us-east-1-cfnlogs.txt
taskcat for pvtendpoint.

**Comment by vsnyc** [comment]
Thansk @SuperMohit , do you have taskcat_output's from the other two tests as well - Atlas without PrivateLink or VPC Peering, and Atlas with VPC Peering?

**Comment by SuperMohit** [comment]
> Thansk @SuperMohit , do you have taskcat_output's from the other two tests as well - Atlas without PrivateLink or VPC Peering, and Atlas with VPC Peering?

https://quickstart-pi.s3.amazonaws.com/quickstart-temp/taskcat_outputs.zip

**Review by vsnyc — APPROVED** [review]

**Pull request merged**